src/components/InjectedComponents/PostDummy.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useEffect } from 'react'
+import { usePostInfoDetails } from '../DataSource/usePostInfo'
+import { DefaultTypedMessageRenderer } from './TypedMessageRenderer'
+import { PluginUI } from '../../plugins/plugin'
+import { makeTypedMessageCompound, isTypedMessageSuspended, isTypedMessageKnown } from '../../protocols/typed-message'
+import { useValueRef } from '../../utils/hooks/useValueRef'
+import { currentWholePostVisibilitySettings, WholePostVisibility } from '../../settings/settings'
+
+export interface PostDummyProps {
+    zip?: () => void
+    unzip?: () => void
+}
+
+export function PostDummy(props: PostDummyProps) {
+    const parsedPostContent = usePostInfoDetails('parsedPostContent')
+    const postPayload = usePostInfoDetails('postPayload')
+    const wholePostVisibilitySettings = useValueRef(currentWholePostVisibilitySettings)
+
+    const processedPostMessage = Array.from(PluginUI.values()).reduce(
+        (x, plugin) => (plugin.postMessageProcessor ? plugin.postMessageProcessor(x) : x),
+        parsedPostContent,
+    )
+    const postDummyVisible =
+        // render dummy for all posts
+        wholePostVisibilitySettings === WholePostVisibility.all ||
+        // render dummy for posts which enhanced by plugins
+        (wholePostVisibilitySettings === WholePostVisibility.enhancedOnly &&
+            processedPostMessage.items.some((x) => !isTypedMessageKnown(x))) ||
```

Comment:
This logic isn't seems right to me. The intermediate processor should mark the TypedMessage as "able to enhance" but the `isTypedMessageKnown` is only detecting if the message type is some built-in type.
What if a plugin return a message type `x-my-msg` and it registered a message renderer for it? In this way, it cannot be rendered.

Reply:
The logic is if the typed message is unknown then inject the `<WholePostView />`.